js/reorder/distanceFunctions/tnw.js

```diff
@@ -4,22 +4,19 @@ function calculateTNWDistance(sim1, sim2) {
     return tndDist * -1;
 }
 
-function getTNWScore(sim1, sim2) {
-    // console.log('Calculating TNW score ', simulation1, simulation2);
-    let simulation1 = sim1.events.filter((d) => { return d.on; });
-    let simulation2 = sim2.events.filter((d) => { return d.on; });
+function getTNWScore(sim1, sim2, GAP_SCORE=-1, MATCH_SCORE=1, MAX_OFFSET_PENALTY=-1, dataFilter = (simulation) => simulation.events.filter((d) => { return d.on; }), dataSelector = d => d.t, resultSelector = (matrix) => matrix[matrix.length - 1][(matrix[matrix.length - 1].length - 1)]) {
```

Comment:
This generalizes the function so you can choose what the data looks like. By default, it takes the same parameters as before.